### PR TITLE
[DO NOT MERGE] Coinmonks Wasabi review

### PR DESCRIPTION
**Pull request by MaxHillebrand**
DO NOT MERGE

This is a ready for review draft of an article that will be published at [Coinmonks](https://medium.com/coinmonks) and [CoinCodeCap](https://coincodecap.com). They want to have a review of Wasabi [according to these guidelines](https://medium.com/@gaurav.agrawal/how-to-write-a-review-8e4dbe80df1a).
I would like to ask you for a review, as we do for the general Wasabi documentation, afterwards I will pull the changes locally, and then finalize / send the article.

It might be good to have a different repo for such kind of review, so to not clutter the documentation. But we don't have one [yet], so I will just open a draft PR and close it.

### CODE REVIEW

File: docs/coinmonks.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+You can generate a new set of private keys, or import already existing ones.
+With Wasabi you can receive bitcoins into your full control, and you can send them to someone else without needing permission from any other person or entity.
+There is also a powerful CoinJoin feature to help obfuscate your transaction history.
+You can use Wasabi to manage your hardware wallet, and it even connects to your own full node.
+Of course, Wasabi is libre and open-source, which means you have full control over the software you manage your money with.
+
+Recently, there was a new release of the software that introduced several awesome features.
+This was the culmination of different streams of development efforts over the past year.
+The feedback from the community of users has been overwhelming, and with this release Wasabi has been [downloaded and installed](https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html) well over 100 000 times already.
+The advanced CoinJoin feature has helped its users to reclaim the privacy of several hundred thousand bitcoin, and the liquidity is ever increasing.
```
Comment (yahiheb) [inline docs/coinmonks.md:17]: This sentence should be moved above, to be under the CoinJoin sentence.

File: docs/coinmonks.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Wasabi Wallet?
+
+## Introducing this privacy weapon
```
[the remark this answers is not in the document]
Comment (MaxHillebrand) [reply docs/coinmonks.md:3]:
```suggestion
## Introducing this privacy tool
```

File: docs/coinmonks.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+[Wasabi](https://wasabiwallet.io) is an open-source, non-custodial, privacy-focused Bitcoin wallet.
+It is a piece of software that runs on your computer and helps you to manage your bitcoins.
+Although Wasabi has some very advanced magic under the hood, it is rather easy to use.
+You can generate a new set of private keys, or import already existing ones.
```
[the remark this answers is not in the document]
Comment (MaxHillebrand) [reply docs/coinmonks.md:8]: By typing in the 12/24 recovery words.

File: docs/coinmonks.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+With Wasabi you can receive bitcoins into your full control, and you can send them to someone else without needing permission from any other person or entity.
```
[the remark this answers is not in the document]
Comment (MaxHillebrand) [reply docs/coinmonks.md:9]: non-custodial

[FYI, this is part of the ELI5 docs]

File: docs/coinmonks.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+There is also a powerful CoinJoin feature to help obfuscate your transaction history.
+You can use Wasabi to manage your hardware wallet, and it even connects to your own full node.
+Of course, Wasabi is libre and open-source, which means you have full control over the software you manage your money with.
```
[the remark this answers is not in the document]
Comment (MaxHillebrand) [reply docs/coinmonks.md:12]: I would say it does...
You can see and review it, and change it to whatever you want. You cannot stop me from making changes to my local copy of it. It's not binaries, but raw source code.

File: docs/coinmonks.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+Recently, there was a new release of the software that introduced several awesome features.
+This was the culmination of different streams of development efforts over the past year.
+The feedback from the community of users has been overwhelming, and with this release Wasabi has been [downloaded and installed](https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html) well over 100 000 times already.
```
Comment (MaxHillebrand) [inline docs/coinmonks.md:16]:
```suggestion
The feedback from the community of users has been overwhelming, and with this release Wasabi has been [downloaded and installed](https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html) well [over 100 000 times](https://somsubhra.com/github-release-stats/?username=zksnacks&repository=WalletWasabi) already.
```

File: docs/coinmonks.md (hunk 1 above)
Comment (MaxHillebrand) [reply docs/coinmonks.md:17]:
```suggestion
```

File: docs/coinmonks.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
+# Wasabi Wallet?
+
+## Introducing this privacy weapon
+
+[Wasabi](https://wasabiwallet.io) is an open-source, non-custodial, privacy-focused Bitcoin wallet.
+It is a piece of software that runs on your computer and helps you to manage your bitcoins.
+Although Wasabi has some very advanced magic under the hood, it is rather easy to use.
+You can generate a new set of private keys, or import already existing ones.
+With Wasabi you can receive bitcoins into your full control, and you can send them to someone else without needing permission from any other person or entity.
+There is also a powerful CoinJoin feature to help obfuscate your transaction history.
```
Comment (MaxHillebrand) [inline docs/coinmonks.md:10]:
```suggestion
There is also a powerful CoinJoin feature to help obfuscate your transaction history, this has helped its users to reclaim the privacy of several hundred thousand bitcoin, and the liquidity is ever increasing.
```

File: docs/coinmonks.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (yahiheb) [reply docs/coinmonks.md:8]: I am not this is what people mean when they say import private keys.

File: docs/coinmonks.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (yahiheb) [reply docs/coinmonks.md:9]: saying `receive bitcoins into your full control` doesn't seem correct.

File: docs/coinmonks.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+For example, if Alice has sent you a bitcoin in the past, and now you want to pay her, then of course select the coin that she already knows about.
+But if you do not want the receiver to know about your previous transaction history, then you should send a coin that was coinjoined previously.
+You easily see which of them they are, by the green check-mark shield.
+After selecting the coins, specify the amounts that should be sent, label the observers who know about this transaction, specify a mining fee, verify that everything is alright, type in your password, and then click send.
+Wasabi will securely sign the transaction, and then broadcast it over the Tor network and with a new Tor identity to a random Bitcoin P2P node, as any other full node would do too.
+
+## But wait, there's more
+
+All the above mentioned would make Wasabi already a top-notch Bitcoin wallet, but there are several more important features that make it even better.
+For a long time, you can connect [your own Bitcoin full node](https://docs.wasabiwallet.io/using-wasabi/BitcoinFullNode.html) to Wasabi, and now Wasabi itself has the bitcoind binaries included in the software.
```
Comment (yahiheb) [inline docs/coinmonks.md:72]: Maybe mention that is it Knots?

File: docs/coinmonks.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+This means that any user can now run a Bitcoin full node with only one single click in the Wasabi GUI.
+As of now, Wasabi does not yet use the full node for consensus verification, but the developers are working hard on making this a reality.
+Further, running your own node is not even a privacy improvement, as Wasabi as a light client is already as good as, maybe even better than, running a full node.
```
Comment (yahiheb) [inline docs/coinmonks.md:75]: > maybe even better than, running a full node.

Is it better than a full node or not? and how?

```suggestion
Further, running your own node is not even a privacy improvement, even as a light client Wasabi is already as good as, maybe even better than, running a full node.
```

File: docs/coinmonks.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+
+Wasabi can also be used to privately [manage your hardware wallet](https://docs.wasabiwallet.io/using-wasabi/ColdWasabi.html), or another offline signing device.
+The software can work with only public keys, to query the chain for your coins, and to build unsigned Bitcoin transactions.
+The private keys to these coins are on the hardware wallet, and this can be connected via USB, or even air-gapped via SD cards with ColdCard wallet.
+After the hardware wallet verified and signed the transaction, it can be broadcast privately through the Tor network.
+This is all enabled by the brilliant Hardware Wallet Interface of the Bitcoin Core project.
+
+The recent release also introduced multi wallet support, meaning that multiple Wasabi wallets can be loaded within one instance of the software.
+This make sit fast and easy to load different wallets and send in between them.
```
Comment (yahiheb) [inline docs/coinmonks.md:87]:
```suggestion
This makes it fast and easy to load different wallets and send in between them.
```

File: docs/coinmonks.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+Part of this is also the amazing feature of [coinjoining directly into another wallet](https://docs.wasabiwallet.io/using-wasabi/Daemon.html), like your hardware wallet.
+
+## What's coming up next
+
+This project is far from finished, and there are multiple open source developers continuously working to improve it further.
+The near term roadmap includes integration of PayJoin, a stegnographic CoinJoin technique between one sender and one receiver.
```
Comment (yahiheb) [inline docs/coinmonks.md:93]: I don't think saying `a steganographic CoinJoin technique between one sender and one receiver` explains what PayJoin is.

```suggestion
The near term roadmap includes the integration of PayJoin, a steganographic CoinJoin technique between one sender and one receiver.
```

File: docs/coinmonks.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+Wasabi is already pretty intuitive to use, once you get the hang of it, but the team is working on a complete re-design of the user experience to make it ever more beautiful.
```
Comment (yahiheb) [inline docs/coinmonks.md:94]:
```suggestion
Wasabi is already pretty intuitive to use, once you get the hang of it, but the team is working on a complete re-design of the user experience to make it even more beautiful.
```

File: docs/coinmonks.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+For the long term, there is ongoing research into WabiSabi, a novel CoinJoin technique which aims to enable sending arbitrary values to third party addresses directly in the CoinJoin.
+If the team succeeds to research and implement this, it would be a massive improvement to current CoinJoin techniques.
```
Comment (yahiheb) [inline docs/coinmonks.md:96]:
```suggestion
If the team succeeds to research and implement this, it would be a massive improvement to the current CoinJoin techniques.
```

File: docs/coinmonks.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (MaxHillebrand) [reply docs/coinmonks.md:16]: the 130k includes the signature files - so that would not be really honest.

File: docs/coinmonks.md (hunk 8 above)
Comment (MaxHillebrand) [reply docs/coinmonks.md:72]: might be too advanced and confusing for such an article...

File: docs/coinmonks.md (hunk 9 above)
Comment (MaxHillebrand) [reply docs/coinmonks.md:75]: Wasabi handles Tor better - i.e. using different tor streams for every connection

Comment (yahiheb) [reply docs/coinmonks.md]: Well then you should maybe specify that it is better in that regard, not aver all.

File: docs/coinmonks.md (hunk 8 above)
Comment (yahiheb) [reply docs/coinmonks.md:71]: I don't think it is advanced to say Wasabi has Knots, because most readers most likely will think it is Core when talking about full node (specially after it was add in 1.1.10) which might be misleading.

Comment (MaxHillebrand) [reply docs/coinmonks.md:71]:
```suggestion
For a long time, you can connect [your own Bitcoin full node](https://docs.wasabiwallet.io/using-wasabi/BitcoinFullNode.html) to Wasabi, and now Wasabi itself has the Bitcoin Knots bitcoind binaries included in the software.
```

File: docs/coinmonks.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (yahiheb) [reply docs/coinmonks.md]: what do you mean by this?

[the remark this answers is not in the document]
Comment (MaxHillebrand) [reply docs/coinmonks.md]: somsubra shows the download number of ALL files in the release - however, the wasabi release includes both packages [deb, tar.gz, win, dmg] as well as the pgp signatures of these files. Of course, who ever downloads the signatures is the same user that also downloaded the package - thus if we want to get closer to a "true" estimate of downloaded releases, then we only count the packages, not the signature files.

**Comment by molnard** [comment]
I am not sure about the structure. Did you use this guide provided in email? https://medium.com/@gaurav.agrawal/8e4dbe80df1a

**Comment by MaxHillebrand** [comment]
@molnard, we have an overview, a summary, pros and cons, features, how it works, user experience.

what is missing from the structure of the medium article is a precise comparison to other wallets [don't want to get into the FUD again] and the pricing structure.

What specifically would you change about the structure?

File: docs/coinmonks.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+## Introducing this privacy tool
+
+[Wasabi](https://wasabiwallet.io) is an open-source, non-custodial, privacy-focused Bitcoin wallet.
+It is a piece of software that runs on your computer and helps you to manage your bitcoins.
+Although Wasabi has some very advanced magic under the hood, it is rather easy to use.
+You can generate a new set of private keys, or import already existing ones.
+With Wasabi you can receive bitcoins into your full control, and you can send them to someone else without needing permission from any other person or entity.
+There is also a powerful CoinJoin feature to help obfuscate your transaction history, this has helped its users to reclaim the privacy of several hundred thousand bitcoin, and the liquidity is ever increasing.
+You can use Wasabi to manage your hardware wallet, and it even connects to your own full node.
+Of course, Wasabi is libre and open-source, which means you have full control over the software you manage your money with.
```
Comment (molnard) [inline docs/coinmonks.md:12]:
```suggestion
Wasabi is libre and open-source, which means you can verify and build the software you manage your money with.
```

File: docs/coinmonks.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+
+Recently, there was a new release of the software that introduced several awesome features.
```
Comment (molnard) [inline docs/coinmonks.md:14]: Why is this sentence here? For example what? Or it will be discussed later?

File: docs/coinmonks.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+## It all starts with synchronization
+
+In general, Wasabi is yet another Bitcoin wallet, but what makes it special, is that every single design decision has been made with privacy as the top priority.
+This already starts when [loading a wallet and synchronising it](https://docs.wasabiwallet.io/using-wasabi/WalletLoad.html) with the Bitcoin blockchain.
+Wasabi is the first wallet to implement BIP 158 block filters, which is a new technique for light clients to privately find out how much bitcoin they have.
+The important thing is that Wasabi does not send your extended public key or addresses to any third party server.
+This sensitive financial data never leaves your computer.
+Instead, the back-end server gives your wallet these block filters, and you check locally if a block is relevant to you.
+If yes, then the wallet client downloads this block from a Bitcoin P2P node, just like any other full node.
+This all happens over the Tor network, and with new Tor identities every time, of course.
```
Comment (molnard) [inline docs/coinmonks.md:27]:
```suggestion
This all happens over the Tor network, and with new Tor identities every time.
```

File: docs/coinmonks.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+The feedback from the community of users has been overwhelming, and with this release Wasabi has been [downloaded and installed](https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html) well [over 100 000 times](https://somsubhra.com/github-release-stats/?username=zksnacks&repository=WalletWasabi) already.
+
+## It all starts with synchronization
+
+In general, Wasabi is yet another Bitcoin wallet, but what makes it special, is that every single design decision has been made with privacy as the top priority.
+This already starts when [loading a wallet and synchronising it](https://docs.wasabiwallet.io/using-wasabi/WalletLoad.html) with the Bitcoin blockchain.
+Wasabi is the first wallet to implement BIP 158 block filters, which is a new technique for light clients to privately find out how much bitcoin they have.
+The important thing is that Wasabi does not send your extended public key or addresses to any third party server.
+This sensitive financial data never leaves your computer.
+Instead, the back-end server gives your wallet these block filters, and you check locally if a block is relevant to you.
```
Comment (molnard) [inline docs/coinmonks.md:25]:
```suggestion
Instead, the back-end server gives your wallet a constant set of filters, and your wallet check locally if a block is relevant to you. In this way, the used node will only know which block contains the relevant transactions.
```

File: docs/coinmonks.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+When receiving bitcoin on an address, then you will notice two important aspects in the graphical user interface.
+First, the receiving address will be removed, as these should only be used once, [and never reused](https://docs.wasabiwallet.io/why-wasabi/AddressReuse.html)!
+Address reuse is one of the biggest privacy leaks in Bitcoin, as this makes it trivial for anyone to see which coins belong to the same user.
+Second, instead of showing only a total balance of the wallet, there is a list of all the coins that can be spent, these are unspent transaction outputs.
+Every coin is clearly labeled by the observers who know about the coin, as this was specified when generating the address.
+This makes coin control and conscious coin selection very intuitive and helps enormously with privacy.
+
+## CoinJoin to reclaim your privacy
+
+If you don't like that your employer knows about where you spend your money, or that a merchant can find out how much money you have, then you should [CoinJoin your bitcoin](https://docs.wasabiwallet.io/using-wasabi/CoinJoin.html), as this breaks the transaction history.
```
Comment (molnard) [inline docs/coinmonks.md:45]:
```suggestion
If you don't like that your employer knows about where you spend your money, or that a merchant can find out how much money you have, then you should [CoinJoin your bitcoin](https://docs.wasabiwallet.io/using-wasabi/CoinJoin.html), as this breaks the chain of transactions.
```

File: docs/coinmonks.md [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+Because every coin is clearly labeled, you know which one are not yet private, for example when You have bought them at a KYC exchange.
+All you need to do to reclaim your privacy, is select this coin, enter your password, and click the button to enqueue and register in the next CoinJoin.
+The privacy gained by the CoinJoin depends on the number of users who register at this time.
+Every user will get one equal value output, which cannot be linked to the specific inputs.
+Wasabi tries to gather 100 users and thus generate 100 anonymity set with every single round.
+At the time of writing, the minimum amount to CoinJoin is roughly 0.1 bitcoin, and there happens a CoinJoin at least once per hour.
+So have a little patience, within a short while you should have a sufficient anonymity set.
+After the round completes, you will see the fresh coin in the list, with a green check-mark shield to indicate that it has reached your privacy target.
+The latest Wasabi release even enables you to efficiently and privately CoinJoin directly into another wallet, like for example your hardware wallet.
```
Comment (molnard) [inline docs/coinmonks.md:56]: Mention the fees too.

File: docs/coinmonks.md (hunk 15 above)
Comment (MaxHillebrand) [reply docs/coinmonks.md:14]: Yes, this is a remenant of a previous draft where I thought it was about the release specifically. It should be removed.

```suggestion
```

File: docs/coinmonks.md (hunk 17 above)
Comment (MaxHillebrand) [reply docs/coinmonks.md:25]:
```suggestion
Instead, the back-end server gives your wallet a constant set of filters, and your wallet checks locally if a block contains a transaction relevant to you.
```

File: docs/coinmonks.md (hunk 18 above)
Comment (MaxHillebrand) [reply docs/coinmonks.md:45]:
```suggestion
If you don't like that your employer knows about where you spend your money, or that a merchant can find out how much money you have, then you should [CoinJoin your bitcoin](https://docs.wasabiwallet.io/using-wasabi/CoinJoin.html), as this breaks the link between inputs and outputs of a transaction.
```

**Comment by MaxHillebrand** [comment]
Thank you everyone for the review!
